staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/conversion/converter.go

```diff
@@ -100,37 +114,62 @@ func (c *crConverter) ConvertFieldLabel(gvk schema.GroupVersionKind, label, valu
 }
 
 func (c *crConverter) Convert(in, out, context interface{}) error {
-	return c.delegate.Convert(in, out, context)
+	unstructIn, ok := in.(*unstructured.Unstructured)
+	if !ok {
+		return fmt.Errorf("input type %T in not valid for unstructured conversion", in)
+	}
+
+	unstructOut, ok := out.(*unstructured.Unstructured)
+	if !ok {
+		return fmt.Errorf("output type %T in not valid for unstructured conversion", out)
+	}
+
+	outGVK := unstructOut.GroupVersionKind()
+	converted, err := c.ConvertToVersion(unstructIn, outGVK.GroupVersion())
+	if err != nil {
+		return err
+	}
+	unstructuredConverted, ok := converted.(runtime.Unstructured)
+	if !ok {
+		// this should not happened
+		return fmt.Errorf("CR conversion failed")
+	}
+	unstructOut.SetUnstructuredContent(unstructuredConverted.UnstructuredContent())
+	return nil
 }
 
 // ConvertToVersion converts in object to the given gvk in place and returns the same `in` object.
 func (c *crConverter) ConvertToVersion(in runtime.Object, target runtime.GroupVersioner) (runtime.Object, error) {
-	// Run the converter on the list items instead of list itself
+	fromGVK := in.GetObjectKind().GroupVersionKind()
+	toGVK, ok := target.KindForGroupVersionKinds([]schema.GroupVersionKind{fromGVK})
```

Comment:
is this incorrect if `in` is a list? it looks like we are computing `toGVK` based on the list's GVK in that case (which seems like it could be very wrong if the list is a `v1.List`)

previously, the first thing we did in this method was iterate over the list, converting the individual items, and left the list GVK alone

Reply:
hmm, that was true for crConverter#ConvertToVersion, but not webhookConverter#ConvertToVersion. I'm having a hard time following the paths that led to the webhook conversion call previously. Does the current state of this PR mean we won't send lists to the webhook to convert, only individual items? Do we have an integration test that exercises the conversion path that gets sent a list?

Reply:
Extended the e2e case on the list to check if the webhook called on the list as whole instead of item by item.